### PR TITLE
[DBX-67288] add exhaustion logging for 526 jobs part 2

### PR DESCRIPTION
**Pull request by SamStuckey**
## Summary

- add / update / homogenize on-exhaustion hooks for 526 related sidekiq async jobs.  This allows for logs that can be aggregated on in Datadog, as well as the addition of stats D metrics.
- fix existing syntax issue with `Rails.logger` vs `::Rails.logger`

### NOTE: PR is part 2 of 2
- [other PR](https://github.com/department-of-veterans-affairs/vets-api/pull/14426)

## Related issue(s)
- [ticket](https://app.zenhub.com/workspaces/disability-benefits-experience-team-carbs-6470c8bfffee9809b2634a52/issues/gh/department-of-veterans-affairs/va.gov-team/67288)

## Testing done

- added RSpec tests
- manual click through and instantiation

## What areas of the site does it impact?
- 526 Async submission

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [x]  I added a screenshot of the developed feature

### CODE REVIEW

**Comment by freeheeling** [comment]
@SamStuckey – To bypass the failing check, you may need to precede the `upsert` on line 167 in `lib/sidekiq/form526_job_status_tracker/job_tracker.rb` with `# rubocop:disable Rails/SkipsModelValidations`. That disablement was removed on line 56 in the previous version.

**Review by ericboehs — APPROVED** [review]

File: lib/sidekiq/form526_backup_submission_process/submit.rb
Comment (sethdarragile6) [inline lib/sidekiq/form526_backup_submission_process/submit.rb:84]: thank you 🤣

**Review by sethdarragile6 — APPROVED** [review]

**Review by freeheeling — APPROVED** [review]

**Review by tblackwe — APPROVED** [review]

**Review by evansmith — APPROVED** [review]

**Review by cilestin — APPROVED** [review]
Blanket approval since Codeowners seems to be wrong again, and requires approval from a team that doesn't actually do anything with this code... 😬

**Pull request merged**
